clients/proxy_pool_registry.go

```diff
@@ -0,0 +1,114 @@
+package clients
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/erpc/erpc/common"
+	"github.com/rs/zerolog"
+)
+
+// contains a set of http.Clients (each configured with a different proxy).
+type ProxyPool struct {
+	ID      string
+	clients []*http.Client
+	counter uint64 // Add atomic counter for round-robin
+}
+
+// returns a round-robin client from the pool.
+func (p *ProxyPool) GetClient() *http.Client {
+	if len(p.clients) == 0 {
+		// Fallback: if for some reason the pool has zero clients
+		return &http.Client{Timeout: 60 * time.Second}
+	}
+	// Round-robin client selection
+	idx := atomic.AddUint64(&p.counter, 1) % uint64(len(p.clients))
+	return p.clients[idx]
+}
+
+// holds all pools
+type ProxyPoolRegistry struct {
+	logger *zerolog.Logger
+	cfg    []common.ProxyPoolConfig
+	pools  sync.Map
+}
+
+// creates a new registry and initializes each pool
+func NewProxyPoolRegistry(
+	cfg []common.ProxyPoolConfig,
+	logger *zerolog.Logger,
+) (*ProxyPoolRegistry, error) {
+	r := &ProxyPoolRegistry{
+		logger: logger,
+		cfg:    cfg,
+	}
+	if len(cfg) == 0 {
+		r.logger.Warn().Msg("no proxy pools defined; all requests will go direct")
+		return r, nil
+	}
+
+	// Initialize each proxy pool
+	for _, poolCfg := range cfg {
+		pool, err := createProxyPool(poolCfg)
+		if err != nil {
+			return nil, err
+		}
+		r.pools.Store(poolCfg.ID, pool)
```

Comment:
Since we're writing only during initialization I don't think sync.Map is needed, it just adds overhead?

Reply:
yeah I think so. we just write to this in the initialization phase, so a basic lock should be enough there